src/Features/CSharp/Portable/CodeLens/CSharpCodeLensMemberFinder.cs

```diff
@@ -0,0 +1,96 @@
+internal class CSharpCodeLensMemberFinder : ICodeLensMemberFinder
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpCodeLensMemberFinder()
+    {
+    }
+
+    public async Task<ImmutableArray<CodeLensMember>> GetCodeLensMembersAsync(Document document, CancellationToken cancellationToken)
+    {
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        using var _ = ArrayBuilder<CodeLensMember>.GetInstance(out var codeLensNodes);
+        var visitor = new CSharpCodeLensVisitor(codeLensNodes.Add);
+
+        visitor.Visit(root);
+
+        return codeLensNodes.ToImmutable();
+    }
+
+    private class CSharpCodeLensVisitor : CSharpSyntaxWalker
+    {
+        private readonly Action<CodeLensMember> _memberFoundAction;
+
+        public CSharpCodeLensVisitor(Action<CodeLensMember> memberFoundAction)
+        {
+            _memberFoundAction = memberFoundAction;
+        }
+
+        public override void VisitClassDeclaration(ClassDeclarationSyntax node)
+        {
+            _memberFoundAction(new CodeLensMember(node, node.Identifier.Span));
+            base.VisitClassDeclaration(node);
+        }
+
+        public override void VisitInterfaceDeclaration(InterfaceDeclarationSyntax node)
+        {
+            _memberFoundAction(new CodeLensMember(node, node.Identifier.Span));
+            base.VisitInterfaceDeclaration(node);
+        }
+
+        public override void VisitEnumDeclaration(EnumDeclarationSyntax node)
+        {
+            _memberFoundAction(new CodeLensMember(node, node.Identifier.Span));
+            base.VisitEnumDeclaration(node);
+        }
+
+        public override void VisitPropertyDeclaration(PropertyDeclarationSyntax node)
+        {
+            _memberFoundAction(new CodeLensMember(node, node.Identifier.Span));
+            base.VisitPropertyDeclaration(node);
+        }
+
+        public override void VisitMethodDeclaration(MethodDeclarationSyntax node)
+        {
+            _memberFoundAction(new CodeLensMember(node, node.Identifier.Span));
+            base.VisitMethodDeclaration(node);
+        }
+
+        public override void VisitStructDeclaration(StructDeclarationSyntax node)
+        {
+            _memberFoundAction(new CodeLensMember(node, node.Identifier.Span));
+            base.VisitStructDeclaration(node);
+        }
+
+        public override void VisitConstructorDeclaration(ConstructorDeclarationSyntax node)
+        {
+            _memberFoundAction(new CodeLensMember(node, node.Identifier.Span));
+            base.VisitConstructorDeclaration(node);
+        }
+
+        public override void VisitRecordDeclaration(RecordDeclarationSyntax node)
+        {
+            _memberFoundAction(new CodeLensMember(node, node.Identifier.Span));
+            base.VisitRecordDeclaration(node);
+        }
```

Comment:
do we not vist events?

Reply:
indeed we do not
<img width="293" alt="image" src="https://user-images.githubusercontent.com/5749229/214703187-a755aa1f-99f2-4847-9e99-52d88a45b63d.png">